health/health.go

```diff
@@ -453,6 +455,14 @@ func (t *Tracker) setHealthyLocked(w *Warnable) {
 	}
 }
 
+// notifyWatchersLocked calls each watcher with nil parameters to signal that
+// control health messages have changed (and should be fetched via CurrentState).
+func (t *Tracker) notifyWatchersLocked() {
+	for _, cb := range t.watchers {
+		cb(nil, nil)
```

Comment:
I wonder if we should change the `watchers` callback signature to be a type with different fields to indicate particular kinds of change, e.g:

```go
type HealthChanged struct {
    StateChanged bool
    WarnableChanged struct { *Warnable; *UnhealthyState }
}
```

I think that might help alleviate newcomer confusion about the semantics of `nil, nil`